// Copyright 2016 The Cartographer Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package cartographer.mapping.scan_matching.proto;

import "cartographer/mapping/proto/scan_matching/icp_scan_matcher_options_2d.proto";

message GlobalICPScanMatcherOptions2D {
  int32 num_global_samples_per_sq_m = 1;
  int32 num_global_rotations = 2;

  double proposal_point_inlier_threshold = 3;
  double proposal_feature_inlier_threshold = 4;

  double proposal_min_points_inlier_fraction = 5;
  double proposal_min_features_inlier_fraction = 6;

  double proposal_features_weight = 7;
  double proposal_points_weight = 8;

  double proposal_raytracing_max_error = 9;

  double proposal_max_points_error = 10;
  double proposal_max_features_error = 11;
  double proposal_max_error = 12;

  int32 min_cluster_size = 13;
  int32 max_cluster_size = 14;
  double min_cluster_distance = 15;

  int32 num_local_samples = 16;

  double local_sample_linear_distance = 17;
  double local_sample_angular_distance = 18;
  
  mapping.scan_matching.proto.ICPScanMatcherOptions2D icp_options = 19;
}
